core/src/aaa/main/screens/MainScreen.java

```diff
@@ -0,0 +1,241 @@
+        borderDOWN = createBox(0, (float) -BORDER_HEIGHT /2, BORDER_WIDTH, 1, true);
+        borderLEFT = createBox((float) -BORDER_WIDTH /2, 0, 1, BORDER_HEIGHT, true);
+        borderRIGHT = createBox((float) BORDER_WIDTH /2, 0, 1, BORDER_HEIGHT, true);
+
+        Label.LabelStyle labelStyle = new Label.LabelStyle();
+        labelStyle.fontColor = Color.BLUE;
+        labelStyle.font = game.font;
+        Label mainText = new Label("Main Screen", labelStyle);
+        mainText.setPosition(450f, 400f);
+        
+        stage.addActor(mainText);
+
+        stage.addListener(new InputListener() {
+            @Override
+            public boolean keyDown(InputEvent ev, int keycode) {
+                if (keycode == com.badlogic.gdx.Input.Keys.ESCAPE) {
+                    game.gameState.paused = true;
+                }
+                return true;
+            }
+        });
+
+        pauseMenu = new PauseMenu(game);
+    }
+
+    @Override
+    public void show() {
+        Gdx.input.setInputProcessor(stage);
+    }
+
+    @Override
+    public void render(float delta) {
+        update(Gdx.graphics.getDeltaTime());
+
+        Gdx.gl.glClearColor(0, 0, 0, 1);
+        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
+
+        //for future use when debugging is needed
+        b2dr.render(world, camera.combined.scl(PPM));
+
+        if (game.gameState.paused && !pauseOpened) {
+            Gdx.input.setInputProcessor(pauseMenu.getStage());
+            pauseOpened = true;
+        } else if (!game.gameState.paused && pauseOpened) {
+            Gdx.input.setInputProcessor(stage);
+            pauseOpened = false;
+        }
+        game.batch.begin();
+        stage.draw();
+        if (game.gameState.paused) {
+            this.pauseMenu.getStage().draw();
+        }
+        game.batch.end();
+    }
+
+    @Override
+    public void resize(int width, int height) {
+        camera.setToOrtho(false, (float) width / SCALE, (float) height / SCALE);
+//        viewport.update(width, height);
+    }
+
+    @Override
+    public void pause() {
+
+    }
+
+    @Override
+    public void resume() {
+
+    }
+
+    @Override
+    public void hide() {
+        Gdx.input.setInputProcessor(null);
+    }
+
+    @Override
+    public void dispose() {
+        world.dispose();
+        b2dr.dispose();
+    }
+
+    //Non tick based renderings
+    public void update(float delta) {
+        world.step(1/60f, 6, 2);
+        inputUpdate(delta);
+        cameraUpdate(delta);
+
+    }
+
+    public void inputUpdate(float delta) {
+        //There are 3 main ways of using the camera,
+        //1. movement based on x,y position rather than physics (can cause clipping)
+        //2. movement based on physics with momentum
+        //3. movement based on physics without momentum
+
+        final int CameraChoice = 3;
+
+        if (CameraChoice == 1) {
+            //Movement based on x,y position rather than physics (can cause clipping)
```

Comment:
it might be better to handle inputs with input processors and Gdx.input.setInputProcessor, and then store things like position in GameState.java, updating the rendering based on that - since it will make those variables accessible elsewhere in the game logic if we need them. Considering MainScreen already has an input processor tied to the stage, you could just use stage.addListener with an overridden InputListener - see lines 73-81. This will complicate things slightly but it should produce cleaner code in the long run.

Reply:
see also https://libgdx.com/wiki/input/event-handling
I think the way I am suggesting should keep the actual position update logic (i.e the code below) in this function and more or less the same, aside from integrating with GameState and InputProcessor